apps/condo/domains/acquiring/schema/RegisterMultiPaymentForInvoicesService.js

```diff
@@ -232,6 +234,19 @@ const RegisterMultiPaymentForInvoicesService = new GQLCustomSchema('RegisterMult
                     implicitFee: Big('0.0'),
                 })
 
+                const amountToPay = Big(totalAmount.amountWithoutExplicitFee)
+                    .add(Big(totalAmount.explicitServiceCharge))
+                    .add(Big(totalAmount.explicitFee))
+                if (acquiringIntegration.minimumPaymentAmount && Big(amountToPay).lt(acquiringIntegration.minimumPaymentAmount)) {
+                    throw new GQLError({
+                        ...ERRORS.PAYMENT_AMOUNT_LESS_THAN_MINIMUM,
+                        messageInterpolation: { minimumPaymentAmount: Big(acquiringIntegration.minimumPaymentAmount).toFixed(2) },
+                    }, context)
+                }
+
+                const payments = await Promise.all(paymentCreateInputs.map((paymentInput) => Payment.create(context, paymentInput)))
```

Comment:
it seems the create operation is asynchronous. So without await there will be an array of promises in the pending state
```suggestion
                const payments = await Promise.all(paymentCreateInputs.map(async (paymentInput) => await Payment.create(context, paymentInput)))
```

Reply:
oh no, everything is correct here. The operation is asynchronous, so there will always be a promise here, and PromiseAll works with promises and will wait for them to complete by default